### ISSUE TITLE
Can we get rid of `permuted=TRUE`? 

### ISSUE THREAD

**Issue by jgabry** [stan-dev/rstan#345]
#### Summary:

The default settings for `extract` and `as.matrix` lead to the draws being returned in different orders. This is because `permuted` defaults to `TRUE` for `extract` but `as.matrix` calls `extract` internally with `permuted=FALSE`. Why are we even permuting the draws at all with `extract`?

**Comment by sakrejda** [comment]
+1 you can't permute until after you diagnose so you'll always call extract with `permuted=FALSE` first anyway.

**Comment by jgabry** [comment]
Yeah definitely. So if anything it should default to FALSE, although even
making it an option at all seems unnecessary.

**Comment by bgoodri** [comment]
It was an Andrew thing

**Comment by bob-carpenter** [comment]
I'm OK with changing it.  Do you want to run it by Andrew
or not during a meeting or just change it?
- Bob

**Comment by bgoodri** [comment]
I wouldn't say we should just get rid of the argument or changing the
default because that will break a lot of existing code. But I could see
doing extraction differently in a release that breaks other things.

**Comment by jgabry** [comment]
Maybe we should deprecate the permuted argument in the next release,
keeping the argument and its default for now but issuing a warning that it
will be removed in a future release? This will lead to a warning whenever
extract is called but that's ok in the short term.

**Comment by bgoodri** [comment]
That could make sense if we thought the extract() function will continue to
exist. Ideally, I would like to get rid of it because extract() is also a
function in one of the tidyverse packages and for Stan, we ought to just be
able to do

posterior$beta

But I agree that the draws should not be permuted.

**Comment by jgabry** [comment]
I guess that depends on how long extract will be around for. If we're
talking months then I agree it's probably not worth it. If years then I
think the deprecation is totally worth it.

**Comment by avehtari** [comment]
I'm also in favor of not permuting. Any progress in this issue?

**Comment by ariddell** [comment]
Just reading this now. I'm in favor of getting rid of the ``extract`` method entirely and replacing it with the Python equivalent of what @bgoodri proposes (e.g., ``posterior['beta']``). Not permuting by default sounds fine to me.

**Comment by bgoodri** [comment]
We'll have to discuss this along with all of the other big changes to rstan
next week.

**Comment by ariddell** [comment]
Was a conclusion reached about this?

**Comment by bgoodri** [comment]
The sooner we can get rid of the option, the better.

**Comment by jgabry** [comment]
I think we decided to definitely get rid of it for RStan 3 (or whatever
version the redesign ends up being). I don't think we decided if it's worth
changing the current default behavior in the meantime.

**Comment by avehtari** [comment]
The sooner we can get rid of the option, the better.

**Comment by ariddell** [comment]
Discussion on the forums http://discourse.mc-stan.org/t/replacement-for-permuted-true-rstan-3-pystan-3/1233
